MonoGame.Framework/Platform/Graphics/Texture2D.OpenGL.cs

```diff
@@ -264,6 +264,13 @@ private void PlatformGetData<T>(int level, int arraySlice, Rectangle rect, T[] d
                         Array.Copy(temp, tempStart, data, dataStart, tRectWidth);
                     }
                 }
+
+            //If we want the full texture rect == (0,0,width,height), we dont have to allocate a temp buffer to copy from.
+            else if (rect.X == 0 && rect.Y == 0 && this.width == rect.Width && this.height == rect.Height)
+            {
+                GL.GetTexImage(TextureTarget.Texture2D, level, glFormat, glType, data);
+                GraphicsExtensions.CheckGLError();
+            }
```

Comment:
The indentation of this code does not seem to match the code around it. It makes it difficult to read.

Reply:
@dellis1972 oops! Thanks for checking 👌.
I've pushed the format fix, additionally ive also removed the duplicate code of pixelToT in each branch outside of the branching logic, and also check the pixelToT == 1 to ensure that we can safely retrieve all data without further processing. We want to ensure that if we use this highly simplified condition we have the same number of bytes in the surface data and type of T data given.